File: src/utils/web/config/searchConfig.ts
/**
 * @author sushao
 * @since 0.4.0
 * @version 0.4.0
 * @description 搜索的配置
 * */
export interface SearchConfig {
  /**
   * 搜索结果的每一项 query 路径
   * */
  li: string;
  /**
   * 在搜索结果中获取 小说 id 的 query 路径
   * */
  novelId: string;
  /**
   * 在搜索结果中获取 小说名字 的 query 路径
   * */
  authorName: string;
  /**
   * 最后一章的路径
   * */
  latestChapterId: string;
  /**
   * 更新时间的正则,如果没有则为 false
   * */
  updateTime: string;
  /**
   * 小说图片
   * */
  image: string;
  /**
   * 标签
   * */
  label: string;
  /**
   * 描述
   * */
  desc: string;
  /**
   * 编码方式
   * */
  encoding: string;
}

File: data/src/store/config/search_config.rs
use serde::{Deserialize, Serialize};
#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchConfig {
    pub(crate) li: String,
    #[serde(rename = "novelId")]
    pub(crate) novel_id: String,
    #[serde(rename = "authorName")]
    pub(crate) author_name: String,
    #[serde(rename = "latestChapterId")]
    pub(crate) latest_chapter_id: String,
    #[serde(rename = "updateTime")]
    pub(crate) update_time: String,
    pub(crate) image: String,
    pub(crate) label: String,
    pub(crate) desc: String,
    pub(crate) encoding: String,
}
